src/AppInstallerCLICore/Commands/ResumeCommand.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "AppInstallerRuntime.h"
+#include "CheckpointManager.h"
+#include "Resources.h"
+#include "ResumeCommand.h"
+#include "RootCommand.h"
+#include "Workflows/ResumeFlow.h"
+
+namespace AppInstaller::CLI
+{
+    using namespace std::string_view_literals;
+    using namespace Execution;
+    using namespace Checkpoint;
+
+    std::vector<Argument> ResumeCommand::GetArguments() const
+    {
+        return {
+            Argument::ForType(Execution::Args::Type::ResumeGuid),
+        };
+    }
+
+    Resource::LocString ResumeCommand::ShortDescription() const
+    {
+        return { Resource::String::ResumeCommandShortDescription };
+    }
+
+    Resource::LocString ResumeCommand::LongDescription() const
+    {
+        return { Resource::String::ResumeCommandLongDescription };
+    }
+
+    Utility::LocIndView ResumeCommand::HelpLink() const
+    {
+        return "https://aka.ms/winget-command-resume"_liv;
+    }
+
+    void ResumeCommand::ExecuteInternal(Execution::Context& context) const
+    {
+        context <<
+            Workflow::EnsureSupportForResume;
+
+        if (context.IsTerminated())
+        {
+            return;
+        }
+
+        auto& checkpointManager = CheckpointManager::Instance();
+
+        int rootContextId = checkpointManager.GetFirstContextId();
+        auto resumeContextPtr = context.CreateEmptyContext(rootContextId);
+
+        Context& resumeContext = *resumeContextPtr;
+        auto previousThreadGlobals = resumeContext.SetForCurrentThread();
+
+        std::string commandName = checkpointManager.GetCommandName(rootContextId);
+        std::unique_ptr<Command> commandToResume;
+
+        // Find the command using the command name.
+        AICLI_LOG(CLI, Info, << "Resuming command: " << commandName);
+        for (auto& command : std::make_unique<RootCommand>()->GetCommands())
+        {
+            if (
+                Utility::CaseInsensitiveEquals(commandName, command->Name()) ||
+                Utility::CaseInsensitiveContains(command->Aliases(), commandName)
+                )
+            {
+                commandToResume = std::move(command);
+                break;
+            }
+        }
+
+        THROW_HR_IF_MSG(E_UNEXPECTED, !commandToResume, "Command to resume not found.");
+
+        resumeContext.SetExecutingCommand(commandToResume.get());
+        resumeContext.SetTargetCheckpoint(checkpointManager.GetLastCheckpoint(rootContextId));
+        resumeContext.SetFlags(Execution::ContextFlag::Resume);
+
+        checkpointManager.Checkpoint(resumeContext, Execution::CheckpointFlag::CommandArguments);
+        
+        commandToResume->Execute(resumeContext);
```

Comment:
Because the command to be resumed is actually calling the command itself, I presume this already handles a change in state on the machine? i.e - User started a command with experimental feature for dependencies enabled, first dependency spawned a reboot, upon reboot, dependencies experimental feature was *magically* disabled by some script that ran during the reboot, and then the `winget resume` happens;  Would it fail due to a change in state? Proceed but just not install further dependencies? What if an argument was used originally that was behind an experimental feature and then was disabled? I think that would cause it to error out and show the "need to enable experimental feature for this", but would that also remove the resume from the checkpoint index?